hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/LogRoller.java

```diff
@@ -141,9 +150,14 @@ public void run() {
     while (!server.isStopped()) {
       long now = System.currentTimeMillis();
       checkLowReplication(now);
-      boolean periodic = false;
       if (!rollLog.get()) {
-        periodic = (now - this.lastrolltime) > this.rollperiod;
+        boolean periodic = false;
+        for (RollController controller : wals.values()) {
+          if (controller.needsPeriodicRoll(now)) {
+            periodic = true;
+            break;
+          }
+        }
```

Comment:
This logic is not same as per branch-2 patches.  What is different here? why?

Reply:
In order to be compatible with jdk1.7